### ISSUE TITLE
Bug: [@typescript-eslint/key-spacing] False-positive for class types surrounded by parentheses

### ISSUE THREAD

**Issue by oliveryasuna** [typescript-eslint/typescript-eslint#7469]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.1.6&fileType=.tsx&code=MYGwhgzhAEBiD29oG8BQ1rwNZgJ4H4AuaCAFwCcBLAOwHNoAfaagVwFsAjAU3IG51m8UgHkcBYgAoyVOo2btu5AJT8AvkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Aa0UNuWIEMyzAObowAbXDYciaNAD20SABpJUrFOyQARogBmCxAGE58OU1G6%2B8ZIhUbNfXfhnHT5jPji3VGyAFs5ABNEUUhkT0oyfEgfCABfHwBdSQS4oA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```typescript
class Foo {
  okay?: string | number;
  notOkay?: (string | number);
}
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/key-spacing": [
      "error",
      {
        "beforeColon": false,
        "afterColon": true,
        "mode": "strict"
      }
    ]
  }
}
```

### tsconfig

```jsonc
// N/A.
{}
```

### Expected Result

No rule violation.

### Actual Result

The rule reports a false-positive because the type is wrapped with parentheses. Further, running fix results in:

```typescript
notOkay?: (string | number); --> notOkay?: string | number);
```

**Comment by bradzacher** [comment]
- please fill in the title
- please fill in all of sections

**Comment by oliveryasuna** [comment]
@bradzacher My bad, done.
